File: Backend/BackendTiendaMigrado/backendtienda/src/main/java/com/backend/backendtienda/controller/EmployeeController.java
package com.backend.backendtienda.controller;

import com.backend.backendtienda.dto.EmployeeDTOs.CreateEmployee;
import com.backend.backendtienda.dto.EmployeeDTOs.GetEmployee;
import com.backend.backendtienda.dto.EmployeeDTOs.UpdateEmployee;
import com.backend.backendtienda.service.EmployeeService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping({"/api/Employee", "/api/employee"})
public class EmployeeController {

    private final EmployeeService service;

    public EmployeeController(EmployeeService service) {
        this.service = service;
    }

    @GetMapping
    public ResponseEntity<List<GetEmployee>> get() {
        return ResponseEntity.ok(service.getAll());
    }

    @GetMapping("/{id}")
    public ResponseEntity<GetEmployee> get(@PathVariable Integer id) {
        return ResponseEntity.ok(service.getById(id));
    }

    @PostMapping
    public ResponseEntity<Void> post(@RequestBody CreateEmployee request) {
        service.add(request);
        return ResponseEntity.ok().build();
    }

    @PutMapping
    public ResponseEntity<Void> put(@RequestBody UpdateEmployee request) {
        service.update(request);
        return ResponseEntity.ok().build();
    }

    @DeleteMapping("/{id}")
    public ResponseEntity<Void> delete(@PathVariable Integer id) {
        service.delete(id);
        return ResponseEntity.ok().build();
    }
}

File: Backend/BackendTienda/BackendTienda/Controllers/EmployeeController.cs
using BackendTienda.Feautures.Employees;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace BackendTienda.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class EmployeeController(EmployeeService _service) : ControllerBase
    {
        [HttpGet]
        public async Task<IActionResult> Get()
        {
            var employees = await _service.GetAll();
            return Ok(employees);
        }

        [HttpGet("{Id}")]
        public async Task<IActionResult> Get(int Id)
        {
            var employee = await _service.GetById(Id);
            return Ok(employee);
        }

        [HttpPost]
        public async Task<IActionResult> Post([FromBody] CreateEmployee request)
        {
            await _service.Add(request);
            return Ok();
        }


        [HttpPut]
        public async Task<IActionResult> Put([FromBody] UpdateEmployee request)
        {
            await _service.Update(request);
            return Ok();
        }

        [HttpDelete("{Id}")]
        public async Task<IActionResult> Delete(int Id)
        {
            await _service.Delete(Id);
            return Ok();
        }

    }
}
